### PR TITLE
DAG-527: Only create re-usable sessions when requested

### PR DESCRIPTION
**Pull request by dbradf**

### CODE REVIEW

**Comment by dbradf** [comment]
We have been seeing some issues with long-running services using the evergreen API. I'm pretty sure this is the issue. With load-balancing and deploys, the session contains stale data.

This will allow scripts to continue to share a session if they wish, but for long-running services will create new sessions when used.

File: docs/source/getting_started.rst [hunk 1]
```diff
@@ -98,3 +98,18 @@ It can be created just like an normal ``EvergreenApi``:
 
 After creating a ``RetryingEvergreenApi`` is can be treated just like a normal
 ``EvergreenApi`` object.
+
+Session
+-------
+
+If you are making several calls to the API is a short amount of time, you may wish to create
+a shared session for all the calls rather. This can be done with the `with_session` context manager.
```
Comment (ancostas) [inline docs/source/getting_started.rst:106]: Add "than a new session for each call".

**Review by ancostas — APPROVED** [review]
LGTM modulo a typo.

**Comment by dbradf** [comment]
evergreen merge

**Comment by dbradf** [comment]
evergreen merge

**Pull request merged**